In [1]:
import pandas as pd

# Carga el archivo CSV
df = pd.read_csv('data/sample3.1.csv', encoding='ISO-8859-1')

# Reemplaza los espacios vacíos con NaN
df.replace('', pd.NA, inplace=True)

# Elimina las columnas con más de un 50% de datos faltantes
df.dropna(axis=1, thresh=df.shape[0]*0.5, inplace=True)

# Elimina las filas con más de un 50% de datos faltantes
df.dropna(axis=0, thresh=df.shape[1]*0.5, inplace=True)


df.dropna(inplace=True)
# Tratamiento de valores atípicos para columnas numéricas
for col in df.select_dtypes(include=['float64', 'int64']).columns:
    Q1 = df[col].quantile(0.25)
    Q3 = df[col].quantile(0.75)
    IQR = Q3 - Q1
    lower_bound = Q1 - 1.5 * IQR
    upper_bound = Q3 + 1.5 * IQR

    # Opción 1: Elimina valores atípicos
    df = df[(df[col] >= lower_bound) & (df[col] <= upper_bound)]


# Guarda el dataframe limpio a un nuevo archivo CSV
df.to_csv('data/sample3.1_cleaned.csv', index=False)

df = pd.read_csv('data/sample3.1_cleaned.csv', encoding='ISO-8859-1')

df

Country Region  DataYear  ClassGrade  Gender  Ageyears        Handed  \
0       USA     CA      2012          12    Male      17.0  Right-Handed   
1       USA     CA      2013          12  Female      17.0  Right-Handed   
2       USA     CA      2017          12    Male      17.0  Right-Handed   
3       USA     CA      2019          12  Female      17.0  Right-Handed   
4       USA     CA      2016          12  Female      17.0  Right-Handed   
..      ...    ...       ...         ...     ...       ...           ...   
201     USA     CA      2022          11  Female      16.0  Ambidextrous   
202     USA     CA      2023          11    Male      16.0  Right-Handed   
203     USA     CA      2021          12    Male      17.0  Right-Handed   
204     USA     CA      2012          12    Male      16.0  Right-Handed   
205     USA     CA      2014          11    Male      17.0  Right-Handed   

    Height_cm Footlength_cm Armspan_cm  ... Watching_TV_Hours Paid_Work_Hours  \
0         178            25        178  ...                 2               0   
1         165            24        160  ...                14               0   
2         196         29.68        201  ...                 4               3   
3         168            24        168  ...                 0               0   
4         170            23        168  ...                 0               0   
..        ...           ...        ...  ...               ...             ...   
201      4.11          4.10        7.5  ...                10              10   
202       180         11.67        183  ...                 6               0   
203     172.7         22.86      171.2  ...                 0               0   
204       160            23        151  ...                 9               0   
205       169            26        170  ...                15               0   

    Work_At_Home_Hours Schoolwork_Pressure  Planned_Education_Level  \
0                    2               A lot          Graduate degree   
1                    0                Some          Graduate degree   
2                    2                Some          Graduate degree   
3                    1         Very little          Graduate degree   
4                    5         Very little          Graduate degree   
..                 ...                 ...                      ...   
201                 10                Some          Graduate degree   
202                  2                Some          Graduate degree   
203                  5                Some          Graduate degree   
204                  3               A lot          Graduate degree   
205                  2                Some                    Other   

    Favorite_Music    Superpower  Preferred_Status     Role_Model_Type  \
0              Pop           Fly             Happy            Relative   
1            Other   Freeze time             Happy            Relative   
2          Country           Fly             Happy       Sports person   
3            Other   Freeze time           Healthy     Business person   
4              Pop     Telepathy             Happy            Relative   
..             ...           ...               ...                 ...   
201         Reggae  Invisibility           Healthy              Friend   
202    Rap/Hip hop   Freeze time           Healthy              Friend   
203           Jazz     Telepathy             Happy  Musician or singer   
204          Other   Freeze time             Happy            Relative   
205    Rap/Hip hop  Invisibility            Famous               Other   

          Charity_Donation  
0                   Health  
1                   Health  
2    Arts, culture, sports  
3              Environment  
4                   Health  
..                     ...  
201      International aid  
202      Wildlife, animals  
203            Environment  
204                 Health  
205      International aid  

[206 rows x 60 columns]

In [2]:
df.shape

(206, 60)

In the first dataset we have 206 rows of data after cleaning, so we will need another data sample in order to concatenate them and complete the 300 rows in total (after cleaning)

In [3]:
import pandas as pd

# Carga el archivo CSV
df = pd.read_csv('data/sample3.2.csv', encoding='ISO-8859-1')

# Reemplaza los espacios vacíos con NaN
df.replace('', pd.NA, inplace=True)

# Elimina las columnas con más de un 50% de datos faltantes
df.dropna(axis=1, thresh=df.shape[0]*0.5, inplace=True)

# Elimina las filas con más de un 50% de datos faltantes
df.dropna(axis=0, thresh=df.shape[1]*0.5, inplace=True)


df.dropna(inplace=True)
# Tratamiento de valores atípicos para columnas numéricas
for col in df.select_dtypes(include=['float64', 'int64']).columns:
    Q1 = df[col].quantile(0.25)
    Q3 = df[col].quantile(0.75)
    IQR = Q3 - Q1
    lower_bound = Q1 - 1.5 * IQR
    upper_bound = Q3 + 1.5 * IQR

    # Opción 1: Elimina valores atípicos
    df = df[(df[col] >= lower_bound) & (df[col] <= upper_bound)]


# Guarda el dataframe limpio a un nuevo archivo CSV
df.to_csv('data/sample3.2_cleaned.csv', index=False)

df = pd.read_csv('data/sample3.2_cleaned.csv', encoding='ISO-8859-1')

df

Country Region  DataYear  ClassGrade  Gender  Ageyears        Handed  \
0       USA     CA      2012          12    Male      17.0  Right-Handed   
1       USA     CA      2013          12  Female      17.0  Right-Handed   
2       USA     CA      2017          12    Male      17.0  Right-Handed   
3       USA     CA      2019          12  Female      17.0  Right-Handed   
4       USA     CA      2016          12  Female      17.0  Right-Handed   
..      ...    ...       ...         ...     ...       ...           ...   
201     USA     CA      2022          11  Female      16.0  Ambidextrous   
202     USA     CA      2023          11    Male      16.0  Right-Handed   
203     USA     CA      2021          12    Male      17.0  Right-Handed   
204     USA     CA      2012          12    Male      16.0  Right-Handed   
205     USA     CA      2014          11    Male      17.0  Right-Handed   

    Height_cm Footlength_cm Armspan_cm  ... Watching_TV_Hours Paid_Work_Hours  \
0         178            25        178  ...                 2               0   
1         165            24        160  ...                14               0   
2         196         29.68        201  ...                 4               3   
3         168            24        168  ...                 0               0   
4         170            23        168  ...                 0               0   
..        ...           ...        ...  ...               ...             ...   
201      4.11          4.10        7.5  ...                10              10   
202       180         11.67        183  ...                 6               0   
203     172.7         22.86      171.2  ...                 0               0   
204       160            23        151  ...                 9               0   
205       169            26        170  ...                15               0   

    Work_At_Home_Hours Schoolwork_Pressure  Planned_Education_Level  \
0                    2               A lot          Graduate degree   
1                    0                Some          Graduate degree   
2                    2                Some          Graduate degree   
3                    1         Very little          Graduate degree   
4                    5         Very little          Graduate degree   
..                 ...                 ...                      ...   
201                 10                Some          Graduate degree   
202                  2                Some          Graduate degree   
203                  5                Some          Graduate degree   
204                  3               A lot          Graduate degree   
205                  2                Some                    Other   

    Favorite_Music    Superpower  Preferred_Status     Role_Model_Type  \
0              Pop           Fly             Happy            Relative   
1            Other   Freeze time             Happy            Relative   
2          Country           Fly             Happy       Sports person   
3            Other   Freeze time           Healthy     Business person   
4              Pop     Telepathy             Happy            Relative   
..             ...           ...               ...                 ...   
201         Reggae  Invisibility           Healthy              Friend   
202    Rap/Hip hop   Freeze time           Healthy              Friend   
203           Jazz     Telepathy             Happy  Musician or singer   
204          Other   Freeze time             Happy            Relative   
205    Rap/Hip hop  Invisibility            Famous               Other   

          Charity_Donation  
0                   Health  
1                   Health  
2    Arts, culture, sports  
3              Environment  
4                   Health  
..                     ...  
201      International aid  
202      Wildlife, animals  
203            Environment  
204                 Health  
205      International aid  

[206 rows x 60 columns]

We have 412 rows of data in total, so we can now concatenate the cleaned data in one single sample with 300 rows of data

In [4]:
import pandas as pd

# Carga los archivos CSV
df1 = pd.read_csv('data/sample3.1_cleaned.csv', encoding='ISO-8859-1')
df2 = pd.read_csv('data/sample3.2_cleaned.csv', encoding='ISO-8859-1')


# Concatena los DataFrames
df_concatenado = pd.concat([df1, df2], ignore_index=True)

# Limita el DataFrame concatenado a las primeras 300 filas
df_limitado = df_concatenado.head(300)

# Guarda el DataFrame limitado en un nuevo archivo CSV
df_limitado.to_csv('data/sample3.csv', index=False)

# Imprime la información del DataFrame limitado
#print(f"El archivo limitado tiene {df_limitado.shape[0]} filas y {df_limitado.shape[1]} columnas.")
df = pd.read_csv('data/sample3.csv', encoding='ISO-8859-1')
df

Country Region  DataYear  ClassGrade  Gender  Ageyears        Handed  \
0       USA     CA      2012          12    Male      17.0  Right-Handed   
1       USA     CA      2013          12  Female      17.0  Right-Handed   
2       USA     CA      2017          12    Male      17.0  Right-Handed   
3       USA     CA      2019          12  Female      17.0  Right-Handed   
4       USA     CA      2016          12  Female      17.0  Right-Handed   
..      ...    ...       ...         ...     ...       ...           ...   
295     USA     CA      2020          12  Female      17.0  Right-Handed   
296     USA     CA      2021          11  Female      18.0  Right-Handed   
297     USA     CA      2014          12  Female      16.0  Right-Handed   
298     USA     CA      2021          11  Female      16.0  Right-Handed   
299     USA     CA      2012          12    Male      17.0   Left-Handed   

    Height_cm Footlength_cm Armspan_cm  ... Watching_TV_Hours Paid_Work_Hours  \
0         178            25        178  ...                 2               0   
1         165            24        160  ...                14               0   
2         196         29.68        201  ...                 4               3   
3         168            24        168  ...                 0               0   
4         170            23        168  ...                 0               0   
..        ...           ...        ...  ...               ...             ...   
295       172            24        172  ...                 2               0   
296     170.1          25.5       12.7  ...                 7               0   
297       167            24        157  ...                 2               1   
298       147            23        155  ...                 2               0   
299       202            20       2000  ...                 0               0   

    Work_At_Home_Hours Schoolwork_Pressure  Planned_Education_Level  \
0                    2               A lot          Graduate degree   
1                    0                Some          Graduate degree   
2                    2                Some          Graduate degree   
3                    1         Very little          Graduate degree   
4                    5         Very little          Graduate degree   
..                 ...                 ...                      ...   
295                  1               A lot          Graduate degree   
296                 21                Some          Graduate degree   
297                  5               A lot          Graduate degree   
298                  5                Some          Graduate degree   
299                  0                Some          Graduate degree   

             Favorite_Music      Superpower  Preferred_Status  \
0                       Pop             Fly             Happy   
1                     Other     Freeze time             Happy   
2                   Country             Fly             Happy   
3                     Other     Freeze time           Healthy   
4                       Pop       Telepathy             Happy   
..                      ...             ...               ...   
295  Rhythm and blues (R&B)       Telepathy             Happy   
296                   Other       Telepathy           Healthy   
297                   Other     Freeze time             Happy   
298             Rap/Hip hop  Super strength             Happy   
299           Rock and roll     Freeze time             Happy   

        Role_Model_Type             Charity_Donation  
0              Relative                       Health  
1              Relative                       Health  
2         Sports person        Arts, culture, sports  
3       Business person                  Environment  
4              Relative                       Health  
..                  ...                          ...  
295            Relative                  Environment  
296          Politician                  Env

In [16]:
df.shape

(300, 60)

Now we have the sample of 300 rows of data from California students, cleaned and ready to concatenate with the samples from other states